plugins/org.eclipse.fordiac.ide.export/src/org/eclipse/fordiac/ide/export/utils/DelayedFiles.java

```diff
@@ -163,7 +136,47 @@ public void clear() {
 
 	/** retrieve an iterable of all filenames currently ready for writing */
 	public List<String> getFilenames() {
-		return storage.stream().map(item -> item.getPath().getFileName().toString()).toList();
+		return storage.stream().map(item -> item.path().getFileName().toString()).toList();
+	}
+
+	/**
+	 * compare if the generated data is already written to disk
+	 *
+	 * @param path  file path to be written to
+	 * @param bytes data to be written as a CharSequence
+	 *
+	 * @return data on disk does not match the generated data
+	 */
+	private static boolean hasChangedContent(final Path p, final CharSequence data) {
+		try {
+			return !Files.readString(p).equals(data);
+		} catch (final IOException e) {
+			return true;
+		}
+	}
+
+	/**
+	 * Creats a backup file of the specified file.
+	 *
+	 * @author Gerhard Ebenhofer, gerhard.ebenhofer@profactor.at
+	 *
+	 * @param in the file that should be backuped
+	 *
+	 * @return the backup file
+	 *
+	 * @throws IOException Signals that an I/O exception has occurred.
+	 */
+	private static File createBakFile(final File in) throws IOException {
```

Comment:
as you moved this code from the utils plugin you need to update the copyright header in this file to not loose the copyright information

Reply:
I hope my changes to the copyright header are understandable?

Reply:
In general it is fine as you did. With EPL v2.0 a bit simplified time syntax is now used. Cleaned that in my commit.